slides/07_Data_Fetching.html

```diff
@@ -199,221 +195,149 @@ <h3>
+            <p>
+              &rarr; Effects are executed at least twice in development<br /><small
+                >(not in prod)</small
+              >
+            </p>
+            <p>
+              Therefore it is recommanded to not fetch data with useEffect but
+              rely on a library, so lets do it!
             </p>
           </section>
           <section data-auto-animate>
-            <p>What about <span class="warn">async-await?</span></p>
-
+            <p>Configure TanStack Query</p>
+            <p>
+              <span class="code">npm i @tanstack/react-query</span>
+            </p>
             <pre
               data-id="code2"
-            ><code class="hljs tsx" data-trim data-line-numbers="1"><script type="text/template">
-                useEffect(async () => {
-
-                  const response = await fetch('/user');
-                  if (!response.ok) throw new Error('Could not fetch data!');
-
-                  const user = await response.json();
-                  setUser(user);
-
-                }, []);
+            ><code class="hljs tsx" data-trim data-line-numbers="1|6-8"><script type="text/template">
+              const queryClient = new QueryClient()
+
+              function App() {
+                /* Provide the client to your App */
+                return (
+                  <QueryClientProvider client={queryClient}>
+                    <MyComponent />
+                  </QueryClientProvider>
+                )
+              }
             </script></code></pre>
+          </section>
 
+          <section data-auto-animate>
             <p>
-              <span class="warn">Does not work!</span> &ndash; The effect
-              function is not async.
+              Using <span class="warn">Tanstack Query</span> for data fetching
             </p>
-          </section>
-          <section data-auto-animate>
-            <p>What about <span class="warn">async-await?</span></p>
 
             <pre
               data-id="code2"
-            ><code class="hljs tsx" data-trim data-line-numbers="1,3,9|"><script type="text/template">
-                useEffect(() => {
+            ><code class="hljs tsx" data-trim data-line-numbers="9-10|1-6"><script type="text/template">
+              
+              const fetcher = async (uri: string): Promise => {
+                const response = await fetch(uri);
+                if (!response.ok) throw new Error('Could not fetch data!');
 
-                  (async function() {
-                    const response = await fetch('/user');
-                    if (!response.ok) throw new Error('Could not fetch data!');
-  
-                    const user = await response.json();
-                    setUser(user);
-                  })();
+                return response.json();
+              };
 
-                }, []);
+              function MyComponent() {
+                const { data: user, isLoading, error } = useQuery(['user'], 
```

Comment:
The formatting seems a bit strange here, I'd suggest to put the first parameter ['user'] on a new line as well (like prettier would format it)